### PR TITLE
📑 Firefox bookmarks plugin

### LINKED ISSUE #13: Plugin: Firefox Bookmarks
Similar to the bookmarks plugin for Brave, read the bookmarks from a local Firefox install and open them in a new Firefox tab when a bookmark is selected.

### PR DESCRIPTION
**Pull request by friedow**
This finally adds a first plugin for firefox :tada:. With this you are able to search and open your firefox bookmarks. Bookmarks are sorted by the number of visits.

This closes #13!

### CODE REVIEW

**Comment by friedow** [comment]
For all the Firefox enjoyers around here FYI @a-kenji, @rti, @velramiir, @pinpox ;)

File: client/src/settings.rs [hunk 1]
```diff
@@ -64,6 +64,18 @@ impl Default for ClockPluginSettings {
     }
 }
 
+#[derive(Debug, Deserialize)]
+pub struct FirefoxBookmarksPluginSettings {
+    #[serde(default = "default_true")]
+    pub enable: bool,
```
Comment (a-kenji) [inline client/src/settings.rs:70]: Do we want them `true`, or `false`?

**Review by a-kenji — APPROVED** [review]
LGTM, one small question.

File: client/src/plugin/firefox/bookmarks.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    fn new() -> Self {
+        Self { entries: vec![] }
+    }
+
+    fn update_entries(&mut self) -> anyhow::Result<()> {
+        self.entries.clear();
+
+        let home_directory = std::env::var("HOME")?;
+        let user = std::env::var("USER")?;
+        let bookmarks_file_path = format!("{home_directory}/.mozilla/firefox/{user}/places.sqlite");
```
Comment (a-kenji) [inline client/src/plugin/firefox/bookmarks.rs:38]: For me the path is incorrect: 
`.mozilla/firefox/profiles.ini`: has profiles in them.

and example profile:
```
[Profile0]
Name=default
IsRelative=1
Path=x32jmy0t.default
Default=1
```

In the same directory is a symlink:
`.mozilla/firefox/x32jmy0t.default`

That contains the `places.sqlite`.

**Comment by friedow** [comment]
I've added a parser for the firefox `profiles.ini` file. @a-kenji can you check if that works on your system as well?

File: client/src/settings.rs (hunk 1 above)
Comment (friedow) [reply client/src/settings.rs:70]: That is a good question. I was thinking true, as the brave browser plugins also are enabled by default. But setting it to true would technically be a breaking change. WDYT?

**Comment by a-kenji** [comment]
@friedow,
It seems to work, but my database is be locked.

```
"/home/kenji/.mozilla/firefox/x32jmy0t.default"
2024-05-09T15:30:49.600Z ERROR [firefox_bookmarks] database is locked (code 5)
thread '<unnamed>' panicked at client/src/plugin/utils.rs:18:17:
explicit panic
```

**Comment by friedow** [comment]
Unlucky, then I'll have to copy the database the same way I did with brave. I'm on it ;)

**Comment by friedow** [comment]
I've duplicated the database on plugin start. @a-kenji wanna test again?

**Pull request merged**